### PR TITLE
=str simplify splitWhen and splitAfter's signature.

### PR DESCRIPTION
**Pull request by He-Pin**

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/Stages.scala [hunk 1]
```diff
@@ -211,6 +212,12 @@ private[stream] object Stages {
     override def withAttributes(attributes: Attributes) = copy(attributes = attributes)
   }
 
+  object Split {
```
Comment (hepin1989) [inline akka-stream/src/main/scala/akka/stream/impl/Stages.scala]: Should I use ```final object``` here?
IDEA said ```final``` for object is redundant.

Comment (ktoso) [inline akka-stream/src/main/scala/akka/stream/impl/Stages.scala]: No need for `final`.
We sometimes use it to mark "intent" more than any kind of correctness.

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
LGTM, let's merge after another LGTM

**Comment by rkuhn** [comment]
LGTM

**Comment by rkuhn** [comment]
~~~
[info]     - close data of size 1 *** FAILED *** (51 milliseconds)
[info]       java.lang.AssertionError: assertion failed: expected class akka.stream.testkit.TestSubscriber$OnError, found class akka.stream.testkit.TestSubscriber$OnComplete$
[info]       at scala.Predef$.assert(Predef.scala:179)
[info]       at akka.testkit.TestKitBase$class.expectMsgClass_internal(TestKit.scala:424)
[info]       at akka.testkit.TestKitBase$class.expectMsgType(TestKit.scala:396)
[info]       at akka.testkit.TestKit.expectMsgType(TestKit.scala:718)
[info]       at akka.stream.testkit.TestSubscriber$ManualProbe.expectError(StreamTestKit.scala:326)
[info]       at akka.http.impl.engine.ws.MessageSpec$TestSetup.expectError(MessageSpec.scala:931)
[info]       at akka.http.impl.engine.ws.MessageSpec$$anonfun$1$$anonfun$apply$mcV$sp$39$$anonfun$apply$mcV$sp$50$$anonfun$apply$mcV$sp$52$$anon$39.<init>(MessageSpec.scala:623)
[info]       at akka.http.impl.engine.ws.MessageSpec$$anonfun$1$$anonfun$apply$mcV$sp$39$$anonfun$apply$mcV$sp$50$$anonfun$apply$mcV$sp$52.apply$mcV$sp(MessageSpec.scala:620)
[info]       at akka.http.impl.engine.ws.MessageSpec$$anonfun$1$$anonfun$apply$mcV$sp$39$$anonfun$apply$mcV$sp$50$$anonfun$apply$mcV$sp$52.apply(MessageSpec.scala:620)
[info]       at akka.http.impl.engine.ws.MessageSpec$$anonfun$1$$anonfun$apply$mcV$sp$39$$anonfun$apply$mcV$sp$50$$anonfun$apply$mcV$sp$52.apply(MessageSpec.scala:620)
~~~

**Comment by hepin1989** [comment]
test passed locally,after several run ,I encounter exception below,but not shows up again.
```
akka-http-core-experimental > testOnly akka.http.impl.engine.ws.MessageSpec
[info] MessageSpec:
[info] The Websocket implementation should
[info]   collect messages from frames
[info]     for binary messages
[info]     - for an empty message (660 milliseconds)
[info]     - for one complete, strict, single frame message (57 milliseconds)
[info]     - for a partial frame (68 milliseconds)
[info]     - for a frame split up into parts (63 milliseconds)
[info]     - for a message split into several frames (63 milliseconds)
[info]     - for several messages (76 milliseconds)
[info]     - unmask masked input on the server side (65 milliseconds)
[info]     - unmask masked input on the server side for empty frame (56 milliseconds)
[info]     for text messages
[info]     - empty message (53 milliseconds)
[info]     - decode complete, strict frame from utf8 (48 milliseconds)
[info]     - decode utf8 as far as possible for partial frame (63 milliseconds)
[info]     - decode utf8 with code point split across frames (163 milliseconds)
[info]     - unmask masked input on the server side (71 milliseconds)
[info]     - unmask masked input on the server side for empty frame (58 milliseconds)
[info]   render frames from messages
[info]     for binary messages
[info]     - for a short strict message (54 milliseconds)
[info]     - for a strict message larger than configured maximum frame size (pending)
[info]     - for a streamed message (68 milliseconds)
[info]     - for a streamed message with a chunk being larger than configured maximum frame size (pending)
[info]     - and mask input on the client side (68 milliseconds)
[info]     - and mask input on the client side for empty frame (53 milliseconds)
[info]     for text messages
[info]     - for a short strict message (52 milliseconds)
[info]     - for a strict message larger than configured maximum frame size (pending)
[info]     - for a streamed message (70 milliseconds)
[info]     - for a streamed message don't convert half surrogate pairs naively (166 milliseconds)
[info]     - for a streamed message with a chunk being larger than configured maximum frame size (pending)
[info]     - and mask input on the client side (67 milliseconds)
[info]     - and mask input on the client side for empty frame (46 milliseconds)
[info]   supply automatic low-level websocket behavior
[info]   - respond to ping frames unmasking them on the server side (49 milliseconds)
[info]   - respond to ping frames masking them on the client side (49 milliseconds)
[info]   - respond to ping frames interleaved with data frames (without mixing frame data) (179 milliseconds)
[info]   - don't respond to unsolicited pong frames (153 milliseconds)
[info]   provide close behavior
[info]   - after receiving regular close frame when idle (user closes immediately) (295 milliseconds)
[info]   - after receiving close frame without close code (55 milliseconds)
[info]   - after receiving regular close frame when idle (user still sends some data) (71 milliseconds)
[info]   - after receiving regular close frame when fragmented message is still open (89 milliseconds)
[info]   - after receiving error close frame with close code and without reason (54 milliseconds)
[info]   - after receiving error close frame with close code and with reason (52 milliseconds)
[info]   - after peer closes connection without sending a close frame (52 milliseconds)
[info]   - when user handler closes (simple) (154 milliseconds)
[info]   - when user handler closes main stream and substream only afterwards (270 milliseconds)
[ERROR] [10/27/2015 11:19:26.910] [MessageSpec-akka.test.stream-dispatcher-8] [ActorSystem(MessageSpec)] Websocket handler failed with Oops, user handler failed!
java.lang.RuntimeException: Oops, user handler failed!
	at akka.http.impl.engine.ws.MessageSpec$$anonfun$1$$anonfun$apply$mcV$sp$39$$anonfun$apply$mcV$sp$49$$anon$37.<init>(MessageSpec.scala:598)
	at akka.http.impl.engine.ws.MessageSpec$$anonfun$1$$anonfun$apply$mcV$sp$39$$anonfun$apply$mcV$sp$49.apply$mcV$sp(MessageSpec.scala:597)
	at akka.http.impl.engine.ws.MessageSpec$$anonfun$1$$anonfun$apply$mcV$sp$39$$anonfun$apply$mcV$sp$49.apply(MessageSpec.scala:597)
	at akka.http.impl.engine.ws.MessageSpec$$anonfun$1$$anonfun$apply$mcV$sp$39$$anonfun$apply$mcV$sp$49.apply(MessageSpec.scala:597)
	at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
	at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
	at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
	at org.scalatest.Transformer.apply(Transformer.scala:22)
	at org.scalatest.Transformer.apply(Transformer.scala:20)
	at org.scalatest.FreeSpecLike$$anon$1.apply(FreeSpecLike.scala:373)
	at org.scalatest.Suite$class.withFixture(Suite.scala:1122)
	at org.scalatest.FreeSpec.withFixture(FreeSpec.scala:1740)
	at org.scalatest.FreeSpecLike$class.invokeWithFixture$1(FreeSpecLike.scala:370)
	at org.scalatest.FreeSpecLike$$anonfun$runTest$1.apply(FreeSpecLike.scala:382)
	at org.scalatest.FreeSpecLike$$anonfun$runTest$1.apply(FreeSpecLike.scala:382)
	at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
	at org.scalatest.FreeSpecLike$class.runTest(FreeSpecLike.scala:382)
	at org.scalatest.FreeSpec.runTest(FreeSpec.scala:1740)
	at org.scalatest.FreeSpecLike$$anonfun$runTests$1.apply(FreeSpecLike.scala:441)
	at org.scalatest.FreeSpecLike$$anonfun$runTests$1.apply(FreeSpecLike.scala:441)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:413)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
	at scala.collection.immutable.List.foreach(List.scala:318)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
	at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:390)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:427)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
	at scala.collection.immutable.List.foreach(List.scala:318)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
	at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:390)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:427)
	at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
	at scala.collection.immutable.List.foreach(List.scala:318)
	at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
	at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:396)
	at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:483)
	at org.scalatest.FreeSpecLike$class.runTests(FreeSpecLike.scala:441)
	at org.scalatest.FreeSpec.runTests(FreeSpec.scala:1740)
	at org.scalatest.Suite$class.run(Suite.scala:1424)
	at org.scalatest.FreeSpec.org$scalatest$FreeSpecLike$$super$run(FreeSpec.scala:1740)
	at org.scalatest.FreeSpecLike$$anonfun$run$1.apply(FreeSpecLike.scala:486)
	at org.scalatest.FreeSpecLike$$anonfun$run$1.apply(FreeSpecLike.scala:486)
	at org.scalatest.SuperEngine.runImpl(Engine.scala:545)
	at org.scalatest.FreeSpecLike$class.run(FreeSpecLike.scala:486)
	at akka.http.impl.engine.ws.MessageSpec.org$scalatest$BeforeAndAfterAll$$super$run(MessageSpec.scala:17)
	at org.scalatest.BeforeAndAfterAll$class.liftedTree1$1(BeforeAndAfterAll.scala:257)
	at org.scalatest.BeforeAndAfterAll$class.run(BeforeAndAfterAll.scala:256)
	at akka.http.impl.engine.ws.MessageSpec.run(MessageSpec.scala:17)
	at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:466)
	at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:677)
	at sbt.TestRunner.runTest$1(TestFramework.scala:84)
	at sbt.TestRunner.run(TestFramework.scala:94)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:219)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:219)
	at sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:207)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:219)
	at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:219)
	at sbt.TestFunction.apply(TestFramework.scala:224)
	at sbt.Tests$.sbt$Tests$$processRunnable$1(Tests.scala:211)
	at sbt.Tests$$anonfun$makeSerial$1.apply(Tests.scala:217)
	at sbt.Tests$$anonfun$makeSerial$1.apply(Tests.scala:217)
	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:45)
	at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:45)
	at sbt.std.Transform$$anon$4.work(System.scala:64)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:237)
	at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:18)
	at sbt.Execute.work(Execute.scala:244)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:237)
	at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:237)
	at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:160)
	at sbt.CompletionService$$anon$2.call(CompletionService.scala:30)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)

[info]   - if user handler fails (164 milliseconds)
[info]     if peer closes with invalid close frame
[info]     - close code outside of the valid range (55 milliseconds)
[info]     - close data of size 1 (54 milliseconds)
[info]     - close message is invalid UTF8 (51 milliseconds)
[info]   - timeout if user handler closes and peer doesn't send a close frame (161 milliseconds)
[info]   - timeout after we close after error and peer doesn't send a close frame (50 milliseconds)
[info]   - ignore frames peer sends after close frame *** FAILED *** (80 milliseconds)
[info]     1002 did not equal 1000 (MessageSpec.scala:920)
[info]     org.scalatest.exceptions.TestFailedException:
[info]     at org.scalatest.MatchersHelper$.newTestFailedException(MatchersHelper.scala:160)
[info]     at org.scalatest.Matchers$AnyShouldWrapper.shouldEqual(Matchers.scala:6303)
[info]     at akka.http.impl.engine.ws.MessageSpec$TestSetup.expectCloseCodeOnNetwork(MessageSpec.scala:920)
[info]     at akka.http.impl.engine.ws.MessageSpec$$anonfun$1$$anonfun$apply$mcV$sp$39$$anonfun$apply$mcV$sp$56$$anon$43.<init>(MessageSpec.scala:669)
[info]     at akka.http.impl.engine.ws.MessageSpec$$anonfun$1$$anonfun$apply$mcV$sp$39$$anonfun$apply$mcV$sp$56.apply$mcV$sp(MessageSpec.scala:662)
[info]     at akka.http.impl.engine.ws.MessageSpec$$anonfun$1$$anonfun$apply$mcV$sp$39$$anonfun$apply$mcV$sp$56.apply(MessageSpec.scala:662)
[info]     at akka.http.impl.engine.ws.MessageSpec$$anonfun$1$$anonfun$apply$mcV$sp$39$$anonfun$apply$mcV$sp$56.apply(MessageSpec.scala:662)
[info]     at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
[info]     at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]     at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]     at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]     at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]     at org.scalatest.FreeSpecLike$$anon$1.apply(FreeSpecLike.scala:373)
[info]     at org.scalatest.Suite$class.withFixture(Suite.scala:1122)
[info]     at org.scalatest.FreeSpec.withFixture(FreeSpec.scala:1740)
[info]     at org.scalatest.FreeSpecLike$class.invokeWithFixture$1(FreeSpecLike.scala:370)
[info]     at org.scalatest.FreeSpecLike$$anonfun$runTest$1.apply(FreeSpecLike.scala:382)
[info]     at org.scalatest.FreeSpecLike$$anonfun$runTest$1.apply(FreeSpecLike.scala:382)
[info]     at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[info]     at org.scalatest.FreeSpecLike$class.runTest(FreeSpecLike.scala:382)
[info]     at org.scalatest.FreeSpec.runTest(FreeSpec.scala:1740)
[info]     at org.scalatest.FreeSpecLike$$anonfun$runTests$1.apply(FreeSpecLike.scala:441)
[info]     at org.scalatest.FreeSpecLike$$anonfun$runTests$1.apply(FreeSpecLike.scala:441)
[info]     at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:413)
[info]     at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
[info]     at scala.collection.immutable.List.foreach(List.scala:318)
[info]     at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]     at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:390)
[info]     at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:427)
[info]     at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
[info]     at scala.collection.immutable.List.foreach(List.scala:318)
[info]     at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]     at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:390)
[info]     at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:427)
[info]     at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
[info]     at scala.collection.immutable.List.foreach(List.scala:318)
[info]     at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]     at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:396)
[info]     at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:483)
[info]     at org.scalatest.FreeSpecLike$class.runTests(FreeSpecLike.scala:441)
[info]     at org.scalatest.FreeSpec.runTests(FreeSpec.scala:1740)
[info]     at org.scalatest.Suite$class.run(Suite.scala:1424)
[info]     at org.scalatest.FreeSpec.org$scalatest$FreeSpecLike$$super$run(FreeSpec.scala:1740)
[info]     at org.scalatest.FreeSpecLike$$anonfun$run$1.apply(FreeSpecLike.scala:486)
[info]     at org.scalatest.FreeSpecLike$$anonfun$run$1.apply(FreeSpecLike.scala:486)
[info]     at org.scalatest.SuperEngine.runImpl(Engine.scala:545)
[info]     at org.scalatest.FreeSpecLike$class.run(FreeSpecLike.scala:486)
[info]     at akka.http.impl.engine.ws.MessageSpec.org$scalatest$BeforeAndAfterAll$$super$run(MessageSpec.scala:17)
[info]     at org.scalatest.BeforeAndAfterAll$class.liftedTree1$1(BeforeAndAfterAll.scala:257)
[info]     at org.scalatest.BeforeAndAfterAll$class.run(BeforeAndAfterAll.scala:256)
[info]     at akka.http.impl.engine.ws.MessageSpec.run(MessageSpec.scala:17)
[info]     at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:466)
[info]     at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:677)
[info]     at sbt.TestRunner.runTest$1(TestFramework.scala:84)
[info]     at sbt.TestRunner.run(TestFramework.scala:94)
[info]     at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:219)
[info]     at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:219)
[info]     at sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:207)
[info]     at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:219)
[info]     at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:219)
[info]     at sbt.TestFunction.apply(TestFramework.scala:224)
[info]     at sbt.Tests$.sbt$Tests$$processRunnable$1(Tests.scala:211)
[info]     at sbt.Tests$$anonfun$makeSerial$1.apply(Tests.scala:217)
[info]     at sbt.Tests$$anonfun$makeSerial$1.apply(Tests.scala:217)
[info]     at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:45)
[info]     at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:45)
[info]     at sbt.std.Transform$$anon$4.work(System.scala:64)
[info]     at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:237)
[info]     at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:237)
[info]     at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:18)
[info]     at sbt.Execute.work(Execute.scala:244)
[info]     at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:237)
[info]     at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:237)
[info]     at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:160)
[info]     at sbt.CompletionService$$anon$2.call(CompletionService.scala:30)
[info]     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]     at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[info]     at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]     at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[info]     at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[info]     at java.lang.Thread.run(Thread.java:745)
[info]   reject unexpected frames
[info]     reserved bits set
[info]     - rsv1 (48 milliseconds)
[info]     - rsv2 (49 milliseconds)
[info]     - rsv3 (52 milliseconds)
[info]   - highest bit of 64-bit length is set (98 milliseconds)
[info]   - control frame bigger than 125 bytes (51 milliseconds)
[info]   - fragmented control frame (46 milliseconds)
[info]   - unexpected continuation frame (52 milliseconds)
[info]   - unexpected data frame when waiting for continuation (52 milliseconds)
[info]   - invalid utf8 encoding for single frame message (54 milliseconds)
[info]   - invalid utf8 encoding for streamed frame (53 milliseconds)
[info]   - truncated utf8 encoding for single frame message (51 milliseconds)
[info]   - truncated utf8 encoding for streamed frame (52 milliseconds)
[info]   - half a surrogate pair in utf8 encoding for a strict frame (51 milliseconds)
[info]   - half a surrogate pair in utf8 encoding for a streamed frame (54 milliseconds)
[info]   - unmasked input on the server side (52 milliseconds)
[info]   - unmasked input on the server side for empty frame (52 milliseconds)
[info]   - masked input on the client side (54 milliseconds)
[info]   - masked input on the client side for empty frame (54 milliseconds)
[info] - support per-message-compression extension (pending)
[info] ScalaTest
[info] Run completed in 5 seconds, 621 milliseconds.
[info] Total number of tests run: 61
[info] Suites: completed 1, aborted 0
[info] Tests: succeeded 60, failed 1, canceled 0, ignored 0, pending 5
[info] *** 1 TEST FAILED ***
[error] Failed: Total 61, Failed 1, Errors 0, Passed 60, Pending 5
[error] Failed tests:
[error] 	akka.http.impl.engine.ws.MessageSpec
[error] (akka-http-core-experimental/test:testOnly) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 6 s, completed 2015-10-27 11:19:28
```

**Comment by rkuhn** [comment]
The MessageSpec failure has been fixed in #18826. Please rebase and then we’ll merge, thanks!

**Comment by hepin1989** [comment]
will rebase then.

**Comment by hepin1989** [comment]
```
[info] RecipeKeepAlive:
Coroner Thread Count starts at 22 in docs.stream.cookbook.RecipeKeepAlive
[info] Recipe for injecting keepalive messages
[info] - must work *** FAILED *** (45 milliseconds)
[info]   java.lang.AssertionError: assertion failed: expected OnNext(ByteString(11)), found OnNext(ByteString(3))
[info]   at scala.Predef$.assert(Predef.scala:179)
[info]   at akka.testkit.TestKitBase$class.expectMsg_internal(TestKit.scala:339)
[info]   at akka.testkit.TestKitBase$class.expectMsg(TestKit.scala:315)
[info]   at akka.testkit.TestKit.expectMsg(TestKit.scala:718)
[info]   at akka.stream.testkit.TestSubscriber$ManualProbe.expectNext(StreamTestKit.scala:262)
[info]   at docs.stream.cookbook.RecipeKeepAlive$$anonfun$1$$anonfun$apply$mcV$sp$1.apply$mcV$sp(RecipeKeepAlive.scala:59)
[info]   at docs.stream.cookbook.RecipeKeepAlive$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(RecipeKeepAlive.scala:12)
[info]   at docs.stream.cookbook.RecipeKeepAlive$$anonfun$1$$anonfun$apply$mcV$sp$1.apply(RecipeKeepAlive.scala:12)
[info]   at org.scalatest.Transformer$$anonfun$apply$1.apply$mcV$sp(Transformer.scala:22)
[info]   at org.scalatest.OutcomeOf$class.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf$.outcomeOf(OutcomeOf.scala:104)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:22)
[info]   at org.scalatest.Transformer.apply(Transformer.scala:20)
[info]   at org.scalatest.WordSpecLike$$anon$1.apply(WordSpecLike.scala:953)
[info]   at org.scalatest.Suite$class.withFixture(Suite.scala:1122)
[info]   at akka.stream.testkit.AkkaSpec.withFixture(AkkaSpec.scala:55)
[info]   at org.scalatest.WordSpecLike$class.invokeWithFixture$1(WordSpecLike.scala:950)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:962)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTest$1.apply(WordSpecLike.scala:962)
[info]   at org.scalatest.SuperEngine.runTestImpl(Engine.scala:306)
[info]   at org.scalatest.WordSpecLike$class.runTest(WordSpecLike.scala:962)
[info]   at akka.stream.testkit.AkkaSpec.runTest(AkkaSpec.scala:55)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1021)
[info]   at org.scalatest.WordSpecLike$$anonfun$runTests$1.apply(WordSpecLike.scala:1021)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:413)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
[info]   at scala.collection.immutable.List.foreach(List.scala:318)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:390)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:427)
[info]   at org.scalatest.SuperEngine$$anonfun$traverseSubNodes$1$1.apply(Engine.scala:401)
[info]   at scala.collection.immutable.List.foreach(List.scala:318)
[info]   at org.scalatest.SuperEngine.traverseSubNodes$1(Engine.scala:401)
[info]   at org.scalatest.SuperEngine.org$scalatest$SuperEngine$$runTestsInBranch(Engine.scala:396)
[info]   at org.scalatest.SuperEngine.runTestsImpl(Engine.scala:483)
[info]   at org.scalatest.WordSpecLike$class.runTests(WordSpecLike.scala:1021)
[info]   at akka.stream.testkit.AkkaSpec.runTests(AkkaSpec.scala:55)
[info]   at org.scalatest.Suite$class.run(Suite.scala:1424)
[info]   at akka.stream.testkit.AkkaSpec.org$scalatest$WordSpecLike$$super$run(AkkaSpec.scala:55)
[info]   at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1067)
[info]   at org.scalatest.WordSpecLike$$anonfun$run$1.apply(WordSpecLike.scala:1067)
[info]   at org.scalatest.SuperEngine.runImpl(Engine.scala:545)
[info]   at org.scalatest.WordSpecLike$class.run(WordSpecLike.scala:1067)
[info]   at akka.stream.testkit.AkkaSpec.org$scalatest$BeforeAndAfterAll$$super$run(AkkaSpec.scala:55)
[info]   at org.scalatest.BeforeAndAfterAll$class.liftedTree1$1(BeforeAndAfterAll.scala:257)
[info]   at org.scalatest.BeforeAndAfterAll$class.run(BeforeAndAfterAll.scala:256)
[info]   at akka.stream.testkit.AkkaSpec.run(AkkaSpec.scala:55)
[info]   at org.scalatest.tools.Framework.org$scalatest$tools$Framework$$runSuite(Framework.scala:466)
[info]   at org.scalatest.tools.Framework$ScalaTestTask.execute(Framework.scala:677)
[info]   at sbt.TestRunner.runTest$1(TestFramework.scala:84)
[info]   at sbt.TestRunner.run(TestFramework.scala:94)
[info]   at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:219)
[info]   at sbt.TestFramework$$anon$2$$anonfun$$init$$1$$anonfun$apply$8.apply(TestFramework.scala:219)
[info]   at sbt.TestFramework$.sbt$TestFramework$$withContextLoader(TestFramework.scala:207)
[info]   at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:219)
[info]   at sbt.TestFramework$$anon$2$$anonfun$$init$$1.apply(TestFramework.scala:219)
[info]   at sbt.TestFunction.apply(TestFramework.scala:224)
[info]   at sbt.Tests$.sbt$Tests$$processRunnable$1(Tests.scala:211)
[info]   at sbt.Tests$$anonfun$makeSerial$1.apply(Tests.scala:217)
[info]   at sbt.Tests$$anonfun$makeSerial$1.apply(Tests.scala:217)
[info]   at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:45)
[info]   at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:45)
[info]   at sbt.std.Transform$$anon$4.work(System.scala:64)
[info]   at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:237)
[info]   at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:237)
[info]   at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:18)
[info]   at sbt.Execute.work(Execute.scala:244)
[info]   at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:237)
[info]   at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:237)
[info]   at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:160)
[info]   at sbt.CompletionService$$anon$2.call(CompletionService.scala:30)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
[info]   at java.util.concurrent.FutureTask.run(FutureTask.java:266)
[info]   at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
[info]   at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
[info]   at java.lang.Thread.run(Thread.java:745)
Coroner Thread Count started at 22, ended at 21, peaked at 27 in docs.stream.cookbook.RecipeKeepAlive
```

**Comment by drewhk** [comment]
PLS BUILD

**Comment by drewhk** [comment]
let's see if recent changes fix the KeepAlive issue

**Comment by rkuhn** [comment]
No, they don't: I left a FIXME to rewrite the cookbook.

**Comment by drewhk** [comment]
Let's just write a stage that does that function better.

**Comment by drewhk** [comment]
Please rebase

**Pull request merged**